tlsutil/config.go

```diff
@@ -321,6 +317,11 @@ func (c *Configurator) OutgoingTLSConfigForCheck(id string) (*tls.Config, error)
 		return nil, err
 	}
 	tlsConfig.InsecureSkipVerify = c.getSkipVerifyForCheck(id)
+	tlsConfig.ServerName = c.base.ServerName
+	if tlsConfig.ServerName == "" {
+		tlsConfig.ServerName = c.base.NodeName
+	}
```

Comment:
OutgoingTLSConfigForChecks is the only place where we allow setting it now. Even that I find questionable... But I don't want to change to many things at once.